Core concepts (#124)
Manta-Network/docs: docs/concepts/ZKP.md

# Zero-Knowledge Proofs

## Origins

The concept of Zero-Knowledge Proofs (ZKPs) was first invented by Shafi Goldwasser, Silvio Micali and Charles Rackoff in their seminal paper, [*The Knowledge Complexity of Interactive-Proof Systems*](https://dl.acm.org/doi/pdf/10.1145/22145.22178) in the 1980s. 

 Despite being considered as a theoretical breakthrough, even the cryptography community labeled the scheme as impossible in practice when the idea was born. Thanks to many breakthroughs made in the recent years, especially the contribution made by many web3 projects like ZCash and Aztec, we have seen a Moore’s Law style improvement on the performance of zero-knowledge proof systems. 

## What is a Zero-Knowledge Proof System?
A zero-knowledge proof system is a protocol by which someone (the prover) can prove the correctness of a statement to someone else (the verifier) without disclosing any additional information. It consists of the following elements and satisfies the following properties.

### Elements of Zero-Knowledge Proof Systems

> * **Statement**: The statement whose veracity we want to proof.

Reviewer: ```suggestion
* **Statement**: The statement whose truth we want to prove.
```

* **Public Input**: The information available to both the prover and the verifier. 
* **Witness**: The information known only by the prover which is enough to prove the statement. The prover wants to keep this information secret from the verifier.

> * **Proof**: A piece of information which is derived by the prover from the statement, public input and witness and can be verified against the statement and the public input to test the veracity of the statement. 

Reviewer: ```suggestion
* **Proof**: A piece of information derived by the prover from the statement, public input, and witness that can be verified against the statement and the public input to test the truth of the claim.
```

### Properties of Zero-Knowledge Proof Systems

Generally, zero-knowledge proof systems must satisfy the following 3 crucial properties:

* **Completeness**: An honest prover can convince the verifier about any statement he/she knows.
* **Soundness**: A computationally bounded prover cannot forfeit a proof that can convince an honest verifier.

> * **Zero-Knowledge**: The proof doesn’t leak any information other than the veracity of the statement itself.

Reviewer: ```suggestion
* **Zero-Knowledge**: The proof doesn’t leak any information other than the truth of the statement itself.
```
